### PR TITLE
Move web.config to application root

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Also reacting to https://github.com/aspnet/Hosting/pull/691.

Review: @Tratcher 

RC2 Approval: @Eilon @DamianEdwards

### CODE REVIEW

**Comment by JunTaoLuo** [comment]
Ignore the "merge into dev" part, this branch is meant to be merged into release then back to dev.

**Comment by moozzyk** [comment]
:ship::it:

File: test/TestSites/web.config [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <system.webServer>
     <!-- This repository cannot use the iis publish tool because there's a bug in the installer if the tool package is a package reference -->
     <!-- See https://github.com/dotnet/cli/issues/1215 -->
-    <aspNetCore forwardWindowsAuthToken="true" processPath="..\TestSites.exe" stdoutLogEnabled="false" stdoutLogFile="..\logs\stdout.log" startupTimeLimit="3600" />
```
Comment (GuardRex) [inline test/TestSites/web.config]: The extension (with timestamp) are automatically created for the log path.

File: test/TestSites/web.config [hunk 2]
```diff
@@ -2,7 +2,7 @@
...
+    <aspNetCore forwardWindowsAuthToken="true" processPath=".\TestSites.exe" stdoutLogEnabled="false" stdoutLogFile=".\logs\stdout.log" startupTimeLimit="3600" />
```
Comment (Tratcher) [inline test/TestSites/web.config]: Remove startupTimeLimit

Comment (moozzyk) [inline test/TestSites/web.config]: why? Isn't it required by tooling (when it arrives?)

Comment (Tratcher) [inline test/TestSites/web.config]: No, we fixed ANCM so it won't time out if the debugger is attached.

Comment (moozzyk) [inline test/TestSites/web.config]: Great! Didn't know about that.

Comment (davidfowl) [inline test/TestSites/web.config]: Why is `.\` required?

Comment (JunTaoLuo) [inline test/TestSites/web.config]: Without it IIS cannot launch. I believe there is an issue filed for that. https://github.com/aspnet/IISIntegration/issues/119

File: samples/IISSample/web.config [hunk 3]
```diff
@@ -4,6 +4,6 @@
     <handlers>
       <add name="aspNetCore" path="*" verb="*" modules="AspNetCoreModule" resourceType="Unspecified" />
     </handlers>
-    <aspNetCore processPath="%DNX_PATH%" arguments="%DNX_ARGS%" stdoutLogEnabled="false" startupTimeLimit="3600" />
```
Comment (muratg) [inline samples/IISSample/web.config]: Why remove the time limit?

Comment (moozzyk) [inline samples/IISSample/web.config]: @muratg https://github.com/aspnet/IISIntegration/pull/124#discussion_r59056705

**Pull request closed without merging**